x/nodes/keeper/valStateChanges.go

```diff
@@ -343,18 +343,26 @@ func (k Keeper) JailValidator(ctx sdk.Ctx, addr sdk.Address) {
 		return
 	}
 	// clear caching for sesssions
-	k.ClearValidatorCache()
+	k.ClearSessionCache()
 	k.deleteValidatorFromStakingSet(ctx, validator)
 	validator.Jailed = true
 	k.SetValidator(ctx, validator)
 	logger := k.Logger(ctx)
 	logger.Info(fmt.Sprintf("validator %s jailed", addr))
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			types.EventTypeJail,
+			sdk.NewAttribute(types.AttributeKeyAddress, addr.String()),
+			sdk.NewAttribute(types.AttributeKeyReason, types.AttributeValueMissingSignature),
+			sdk.NewAttribute(types.AttributeKeyJailed, addr.String()),
```

Comment:
Shouldn't this be `validator.Jailed` instead of `addr.String()`

Reply:
Maybe, this was copied over from legacy cosmos code